### PR TITLE
Make javalab validation code show up in editing start_sources in levelbuilder

### PR DESCRIPTION
**Pull request by JillianK**
This PR adds logic to include validation code as part of the displayed sources in edit start_sources mode in levelbuilder and a check when saving files to ensure that a student's file isn't the same name as a validation code file (not handled by existing logic because validation isn't part of sources in student code).  This is a follow up to [this PR](https://github.com/code-dot-org/code-dot-org/pull/40668)

Here's the levelbuilder version:
Screen Shot 2021-05-18 at 3 11 48 PM

Here's the student facing version:
Screen Shot 2021-05-18 at 3 11 39 PM

Here's the error message when the student tries to name a file to the same name as a validation file:
Screen Shot 2021-05-18 at 3 08 21 PM

Here's the error message when the student tries to name a file to the same name as a support file:
Screen Shot 2021-05-21 at 12 04 17 PM

## Links

- PR that this follows up: [here](https://github.com/code-dot-org/code-dot-org/pull/40668)
- jira ticket: [CSA-313](https://codedotorg.atlassian.net/browse/CSA-313?atlOrigin=eyJpIjoiZmU0NjM3OTAzMWI3NDAwN2E0MzliNWVlNzkyZTYwZjIiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jmkulwik** [comment]
Just to double check: Does the editor populate the code in MyValidation.java? i.e. the screenshots show the filenames, but not file content.

File: apps/src/javalab/JavalabEditor.jsx [hunk 1]
```diff
@@ -399,6 +405,10 @@ class JavalabEditor extends React.Component {
     return `Filename ${filename} is already in use in this project. Please choose a different name`;
   }
 
+  duplicateValidationFileError(filename) {
+    return `Filename ${filename} is already in use in this level's support code. Please choose a different name`;
```
Comment (jmkulwik) [inline apps/src/javalab/JavalabEditor.jsx]: We now have an i18n file for Java Lab :) Go ahead and add this string there.

File: apps/src/javalab/JavalabEditor.jsx [hunk 2]
```diff
@@ -298,6 +299,11 @@ class JavalabEditor extends React.Component {
         renameFileError: this.duplicateFileError(newFilename)
       });
       return;
+    } else if (Object.keys(this.props.validation).includes(newFilename)) {
+      this.setState({
+        renameFileError: this.duplicateValidationFileError(newFilename)
```
Comment (jmkulwik) [inline apps/src/javalab/JavalabEditor.jsx]: Mind adding a task on the backlog to update the styling here for dark mode?

File: apps/src/javalab/JavalabEditor.jsx [hunk 3]
```diff
@@ -298,6 +299,11 @@ class JavalabEditor extends React.Component {
         renameFileError: this.duplicateFileError(newFilename)
```
Comment (jmkulwik) [inline apps/src/javalab/JavalabEditor.jsx]: Looking at the two error messages, the `duplicateValidationFileError` message is also what we'd want if the student named a file after hidden support code. Mind handling that case here as well?

**Comment by JillianK** [comment]
> Just to double check: Does the editor populate the code in MyValidation.java? i.e. the screenshots show the filenames, but not file content.

It does! Here's an updated screenshot showing text:
Screen Shot 2021-05-21 at 12 14 24 PM

**Review by molly-moen — APPROVED** [review]
LGTM!

**Review by jmkulwik — APPROVED** [review]
LGTM!

**Pull request merged**